security-proxy/src/main/java/org/georchestra/security/Proxy.java

```diff
@@ -194,21 +194,37 @@ public HttpUriRequest getRedirect(HttpRequest httpRequest, HttpResponse httpResp
     private Permissions sameDomainPermissions;
     private String proxyPermissionsFile;
 
+    /**
+     * This variable controls the socketTimeout parameter passed to the httpclient
+     * configuration, used for proxified queries. Relying on HttpClient's
+     * documentation
+     * (https://hc.apache.org/httpcomponents-client-4.2.x/tutorial/html/connmgmt.html),
+     * it corresponds to the max time between two consecutive packets, and is
+     * expressed in milliseconds.
+     *
+     */
     private Integer httpClientTimeout = 300000;
 
-    private final static String setCookieHeader = "Set-Cookie";
-    private HttpAsyncClientBuilder httpAsyncClientBuilder;
-
     public void setHttpClientTimeout(Integer timeout) {
         this.httpClientTimeout = timeout;
     }
 
-    public void setPublicUrl(String publicUrl) {
-        this.publicUrl = publicUrl;
+    /**
+     * This variable holds the timeout in minutes before the proxified HTTP requests
+     * will be discarded, throwing a java.util.concurrent.TimeoutException, see
+     * return of executeHttpRequest() method below.
+     */
+    private Integer httpQueryTimeout = 20;
+
+    public void setHttpQueryTimeout(Integer httpQueryTimeout) {
+        this.httpQueryTimeout = httpQueryTimeout;
     }
 
-    public Integer getHttpClientTimeout() {
-        return httpClientTimeout;
+    private final static String setCookieHeader = "Set-Cookie";
+    private HttpAsyncClientBuilder httpAsyncClientBuilder;
+
+    public void setPublicUrl(String publicUrl) {
```

Comment:
Could the public URL be of type `java.net.URL` ?

Reply:
Note: I just moved the code, I did not add this one.

My guess is that it is a string because it comes from the spring XML configuration file (and further from the properties files from the datadir), it seems that this publicUrl is indeed just used here:
https://github.com/georchestra/georchestra/blob/7915e2fa339c584264eea9c96447e5d25fb927f0/security-proxy/src/main/java/org/georchestra/security/Proxy.java#L272-L274
As a result, it could probably be declared as an URL, but since the actual URL object is created from the string only once in the init() method, the modification won't provide any improvements other than a cosmetic one.